### ISSUE TITLE
Error in output variable though aws_secretsmanager_secret resource got created successfully

### ISSUE THREAD

**Issue by prithakundu** [hashicorp/terraform#25331]
### Terraform Version

```
Terraform v0.11.8
```

### Terraform Configuration Files

```hcl
provider "aws" {
  region  = "us-east-1"
  version = "2.43.0"
}

resource "aws_secretsmanager_secret" "ops_els_password" {
  name      = "asc/test/ops/els/password"
  recovery_window_in_days = 0
}

output "ops_els_password_arn" {
  value = "${aws_secretsmanager_secret.ops_els_password.arn}"
}
```

### Expected Behavior
Since the resource got created, there shouldn't be any issue in computing the output variable. However, this is an intermittent issue. 

### Actual Behavior
aws_secretsmanager_secret.ops_els_password: Creation complete after 0s
error occurred:
* output.ops_els_password_arn: Resource 'aws_secretsmanager_secret.ops_els_password' does not have attribute 'arn' for 
variable 'aws_secretsmanager_secret.ops_els_password.arn'

### Steps to Reproduce

1. `terraform init`
2. `terraform apply`

### Additional Context
 This is an intermittent issue. Though the resource got created, in computing output variable its not able to find arn. How to resolve this kind of issue

**Comment by apparentlymart** [comment]
Hi @prithakundu,

I recognize this as a bug from Terraform 0.11 that was fixed in Terraform 0.12. Unfortunately, Terraform 0.11 is no longer maintained, so there will be no fix for this issue in the 0.11 release line. You'll need to [upgrade to Terraform 0.12](https://www.terraform.io/upgrade-guides/0-12.html) to get the fix for this bug, along with fixes for many other configuration language bugs (which was the focus of the Terraform 0.12 release).

Because Terraform 0.11 is no longer maintained, I'm going to close this. If you still see errors like this after upgrading to Terraform 0.12 or later, please do open a new issue and we can investigate it further.

Thanks!

**Comment by richardgavel** [comment]
Do you have any more information about the bug that was fixed?
